### PR TITLE
Work towards an Android testing framework and fix a few bugs

### PR DESCRIPTION
**Pull request by xymus**
Update check_android.sh to accept any APK file, many files at once, deduce the package name from the APK and accept some command line arguments. Users can manage their own emulator (or device) and they can show the emulator window.

Rename mnit_simple to mnit_test with 2 alternatives on Android: minimal and complete. Also fix what was broken in the complete test and the lib. The complete test still doesn't work because of a bug in `android::audio` fixed by #1587 which is not yet merged.

Consider this PR as the first part for integrating Android test in jenkins. It will be easier to tweak the integration with already the script in master.

This PR does not fix the issue with some Android apps where they crash after about 30 seconds, but it could help prevent future regressions.

### CODE REVIEW

**Comment by privat** [comment]
+1
This is great

File: contrib/crazy_moles/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ bin/moles: $(shell ../../bin/nitls -M src/moles_linux.nit) assets/images/drawing
 	mkdir -p bin
 	../../bin/nitc -o bin/moles src/moles_linux.nit
 
-android: android-icons $(shell ../../bin/nitls -M src/moles_android.nit) assets/images/drawing.png
+bin/moles.apk: android-icons $(shell ../../bin/nitls -M src/moles_android.nit) assets/images/drawing.png
```
Comment (privat) [inline contrib/crazy_moles/Makefile]: I suggest that an `android` rule still exists in project that can generate apk files. It is a simple convention for the human and allow mass invocation like #1652 

    android: bin/moles.apk

**Comment by xymus** [comment]
Rerolled for tests and @privat suggestion to keep the `android:` rule in the crazy moles makefile.

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
Alexis, can you explain the reroll? Is this ready to merge?

**Comment by xymus** [comment]
Rerolled to renamed the generated XML file to `android-shortappname.xml`, it's ready to merge.

**Comment by privat** [comment]
I assume I have to run `make check-android` in projects that have one?

File: contrib/crazy_moles/Makefile [hunk 2]
```diff
@@ -15,15 +17,15 @@ android-icons: ../../contrib/inkscape_tools/bin/svg_to_icons art/icon.svg
...
 	adb install -rf bin/moles.apk
 
 assets/images/drawing.png: art/drawing.svg ../../contrib/inkscape_tools/bin/svg_to_icons
 	mkdir -p assets/images
 	../inkscape_tools/bin/svg_to_png_and_nit --src src/ --scale 2.0 art/drawing.svg
 
-check-android:
-	./check-android.sh
+check-android: bin/moles.apk
+	../../misc/jenkins/check-android.sh bin/moles.apk
```
Comment (privat) [inline contrib/crazy_moles/Makefile]: bug, the program is named `check_android.sh`, with an underscore.

**Comment by xymus** [comment]
Rerolled to fix script name in the crazy_moles Makefile.

`make check-android` should work for crazy_moles. You may need to install the the x86 android-19 ABI (or emulator image) for it to work with the default configuration. You can also launch the emulator manually (or connect a physical device), and run the script with `check_android.sh --dev dev-id-from-adb-devices file_to_test.apk`.

**Comment by privat** [comment]
I assume there was a problem in the rebase

**Comment by xymus** [comment]
It should be Ok now.

**Pull request merged**